src/main/java/org/ohdsi/authenticator/service/AuthenticatorProxyMode.java

```diff
@@ -0,0 +1,49 @@
+package org.ohdsi.authenticator.service;
+
+import io.jsonwebtoken.Claims;
+import org.ohdsi.authenticator.exception.AuthenticationException;
+import org.ohdsi.authenticator.model.UserInfo;
+import org.pac4j.core.credentials.Credentials;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AuthenticatorProxyMode implements Authenticator {
+    protected static final Logger logger = LoggerFactory.getLogger(AuthenticatorProxyMode.class.getName());
+
+    private TokenService tokenService;
+    private TokenProvider tokenProvider;
+
+    public AuthenticatorProxyMode(TokenService tokenService,
+                                  TokenProvider tokenProvider) {
+
+        this.tokenService = tokenService;
+        this.tokenProvider = tokenProvider;
+    }
+
+    @Override
+    public UserInfo authenticate(String method, Credentials request) {
+
+        throw new AuthenticationException("authentication is delegated to the proxy level, for PROXY authentication mode");
+    }
+
+    @Override
+    public String resolveUsername(String token) {
+
+        return tokenService.resolveAdditionalInfo(token, Claims.SUBJECT, String.class);
+    }
+
+
+    @Override
+    public UserInfo refreshToken(String token) {
+
+        return tokenService.resolveUser(token);
```

Comment:
yeah, this is a good question. I would say it logical that this method is going to never call for IAP, but there are a few reasons why I did this:
1) I cannot be absolutely sure when/where we call refresh from UI
2) We do the same things for WebAPI call resolveUser instead of refreshing token